src/batch-client/index.ts

```diff
@@ -1477,12 +1480,16 @@ export class ZimbraBatchClient {
 
 			this.checkAndUpdateSessionId(sessionId);
 
-			if (notifications && this.notificationHandler) {
+			if (notifications && this.notificationsEmitter) {
 				// as notification handling happens in synchronous way, if the notifications count is really higher (for bulk operations)
 				// the UI would freeze because of the JavaScript execution time. Hence, delayed the notification handling to give some time
 				// for the render to happen in the JavaScript event loop
 				setTimeout(() => {
-					this.notificationHandler && this.notificationHandler(notifications);
+					this.notificationsEmitter &&
+						this.notificationsEmitter.emit(
```

Comment:
curious, if we are using event emitter then do we need setTimeout wrapping it? as I feel event emitter works on event loop only

Reply:
Yes now the setTimeout would be irrelevant, I will remove it...

Reply:
Removed

Reply:
awesome